articles/node_lifecycle.md

```diff
@@ -0,0 +1,176 @@
+
+### Inactive state
+
+This state represents a node that is not currently performing any processing.
+
+The main purpose of this state is to allow a node to be (re-)configured (changing configuration parameters, adding and removing topic publications/subscriptions, etc.) without altering its behaviour while it is running.
+
+[To discuss: Should there be a callback for when configuration is changed, e.g. altered parameters, new topics, etc.?]
+
+While in this state, the node will not receive any execution time to read topics, perform processing of data, respond to functional service requests, etc.
+
+In the inactive state, any data that arrives on topics will not be read. Whether or not data is queued will depend on the QoS parameters of the topics involved. [In other words, to queue data or not is up to each topic.]
+
+Any service requests to a node in the inactive state will not be answered (to the caller, they will fail immediately).
+
+Entering the `inactive` state from the `created` state requires running the node's `onInitialised` method. The node uses this to set up any resources it must hold throughout its life (irrespective of if it is active or inactive). As examples, such resources may include topic publications and subscriptions, memory that is held continuously, and initialising configuration parameters.
+
+Entering the `inactive` state from the `active` state requires running the node's `onDeactivated` method. See the `active` state description for more information.
+
+Entering the `inactive` state from the `fatal error` state requires running the node's `onReset` method. See the `fatal error` state description for more information.
+
+If an error occurs while in this state that cannot be handled by the node, then the transition to the `fatal error` state is taken.
+
+The node may also exit this state by being destroyed, in which case the `destroy` transition is taken.
+
+Implementation note: Causing the transition to the destroyed state can often be mapped to deletion of the node instance.
+
+### Active state
+
+This is the main state of the node's life cycle. While in this state, the node performs any processing, responds to service requests, reads and processes data, produces output, etc.
+
+Entering the `active` state requires running the node's `onActivated` method. This is used to prepare the node for execution. This may include acquiring resources that are only held while the node is actually active, such as access to hardware. Ideally, no preparation that requires significant time (such as lengthy hardware initialisation) should be performed in this callback.
+
+To shut down a node, it must first be deactivated. This will cause the `onDeactivate` method to be called, allowing the node to clean up any resources it only holds while active.
+
+If an error that cannot be handled by the node/system occurs in this state, the transition to the `fatal error` state is taken.
```

Comment:
Mmmm. On the implementation level, whether one should equate "uncaught C++ exception" and "error" should  be discussed. A lot of people are against exceptions for starters, especially in the embedded / hard RT domain.

NB: it would definitely work for me.